contracts/APRCalculator/modules/MacroFactor/MacroFactor.sol

```diff
@@ -57,18 +52,49 @@ abstract contract MacroFactor is IMacroFactor, Initializable, System, Governed {
 
     // _______________ Internal functions _______________
 
+    /**
+     * @inheritdoc Price
+     * @notice Update the macro factor based on the price update, if the needed conditions are met.
+     * @param _price The price to be used for the update.
+     */
+    function _onPriceUpdate(uint256 _price) internal override(Price) {
+        if (!disabledMacro) {
+            _triggerMacroUpdate(_price);
+        }
+    }
+
+    // _______________ Private functions _______________
+
+    /**
+     * @notice Trigger the macro factor update.
+     * @param _price The price to be used for the update.
+     */
+    function _triggerMacroUpdate(uint256 _price) private {
+        updatedPrices.push(_price);
+        smaFastSum += _price;
+        smaSlowSum += _price;
+
+        uint256 arrLenght = updatedPrices.length;
+        if (arrLenght >= SLOW_SMA) {
```

Comment:
You must ensure that the length of the prices array is enough by adding all previous days' data on initialize